src/modules/hip/kernel/tensor_stddev.hpp

```diff
@@ -0,0 +1,907 @@
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    /* Stores individual channel Variations computed from channel Means to compute Stddev of individual channels*/
+    __shared__ float partialRVar_smem[16][16];                                                   // 16 rows of src, 128 reduced cols of src in a 16 x 16 thread block
+    __shared__ float partialGVar_smem[16][16];
+    __shared__ float partialBVar_smem[16][16];
+    float *partialRVarRowPtr_smem = &partialRVar_smem[hipThreadIdx_y][0];                       // float pointer to beginning of each row in _smem
+    float *partialGVarRowPtr_smem = &partialGVar_smem[hipThreadIdx_y][0];
+    float *partialBVarRowPtr_smem = &partialBVar_smem[hipThreadIdx_y][0];
+    partialRVarRowPtr_smem[hipThreadIdx_x] = 0.0f;                                              // initialization of _smem to 0 using all 16 x 16 threads
+    partialGVarRowPtr_smem[hipThreadIdx_x] = 0.0f;
+    partialBVarRowPtr_smem[hipThreadIdx_x] = 0.0f;
+
+    int index       = id_z * 4;
+    float4 meanR_f4 = (float4)mean[index];
+    float4 meanG_f4 = (float4)mean[index + 1];
+    float4 meanB_f4 = (float4)mean[index + 2];
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    int xAlignedLength = roiTensorPtrSrc[id_z].xywhROI.roiWidth & ~7;                           // alignedLength for vectorized global loads
+    int xDiff = roiTensorPtrSrc[id_z].xywhROI.roiWidth - xAlignedLength;                        // difference between roiWidth and alignedLength
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+
+    d_float24 src_f24, tempCh_f24, tempChSq_f24;
+    rpp_hip_load24_pln3_and_unpack_to_float24_pln3(srcPtr + srcIdx, srcStridesNCH.y, &src_f24); // load 24 pixels to local memory
+    rpp_hip_math_subtract8_const(&src_f24.f8[0], &tempCh_f24.f8[0], meanR_f4);                  // subtract mean from each pixel
+    rpp_hip_math_multiply8(&tempCh_f24.f8[0], &tempCh_f24.f8[0], &tempChSq_f24.f8[0]);          // square the temporary value
+    rpp_hip_math_subtract8_const(&src_f24.f8[1], &tempCh_f24.f8[1], meanG_f4);
+    rpp_hip_math_multiply8(&tempCh_f24.f8[1], &tempCh_f24.f8[1], &tempChSq_f24.f8[1]);
+    rpp_hip_math_subtract8_const(&src_f24.f8[2], &tempCh_f24.f8[2], meanB_f4);
+    rpp_hip_math_multiply8(&tempCh_f24.f8[2], &tempCh_f24.f8[2], &tempChSq_f24.f8[2]);
+
+    if (id_x + 8 > roiTensorPtrSrc[id_z].xywhROI.roiWidth)                                      // local memory reset of invalid values (from the vectorized global load) to 0.0f
+    {
+        for(int i = xDiff; i < 8; i++)
+        {
+            tempChSq_f24.f8[0].f1[i] = 0.0f;
```

Comment:
Please check if it is better to rename 
 tempChSq to tempChannelSquared if that is what it means